### PR TITLE
Decouple external operation handler

### PR DESCRIPTION
**Pull request by geirst**
@vekterli please review
@baldersheim @toregge FYI

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/tests/common/teststorageapp.h [hunk 1]
```diff
@@ -85,6 +87,7 @@ class TestStorageApp
         { return _compReg.getDistribution(); }
     TestNodeStateUpdater& getNodeStateUpdater() { return _nodeStateUpdater; }
     uint16_t getIndex() const { return _compReg.getIndex(); }
+    const NodeIdentity& node_identity() const { return _node_identity; }
```
[the remark this answers is not in the document]
Comment (geirst) [reply storage/src/tests/common/teststorageapp.h]: Fixed

File: storage/src/vespa/storage/distributor/distributor_operation_context.h [hunk 2]
```diff
@@ -31,11 +33,17 @@ class DistributorOperationContext {
...
+    virtual DistributorBucketSpaceRepo& read_only_bucket_space_repo() = 0;
+    virtual document::BucketId make_split_bit_constrained_bucket_id(const document::DocumentId& docId) const = 0;
 
+    virtual const DistributorConfiguration& distributor_config() const = 0;
     virtual void send_inline_split_if_bucket_too_large(document::BucketSpace bucket_space,
                                                        const BucketDatabase::Entry& entry,
                                                        uint8_t pri) = 0;
-    virtual const DistributorConfiguration& distributor_config() const = 0;
+    virtual OperationRoutingSnapshot read_snapshot_for_bucket(const document::Bucket& bucket) const = 0;
+    virtual PendingMessageTracker& pending_message_tracker() = 0;
```
[the remark this answers is not in the document]
Comment (geirst) [reply storage/src/vespa/storage/distributor/distributor_operation_context.h]: Fixed

File: storage/src/vespa/storage/distributor/externaloperationhandler.cpp [hunk 3]
```diff
@@ -29,43 +29,52 @@ LOG_SETUP(".distributor.manager");
...
 
 class DirectDispatchSender : public DistributorMessageSender {
-    Distributor& _distributor;
+    DistributorNodeContext& _node_ctx;
+    NonTrackingMessageSender& _msg_sender;
 public:
-    explicit DirectDispatchSender(Distributor& distributor)
-        : _distributor(distributor)
+    explicit DirectDispatchSender(DistributorNodeContext& node_ctx,
+                                  NonTrackingMessageSender& msg_sender)
```
[the remark this answers is not in the document]
Comment (geirst) [reply storage/src/vespa/storage/distributor/externaloperationhandler.cpp]: Fixed

**Comment by geirst** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**